### PR TITLE
Add windows installing instructions

### PR DESCRIPTION
**Pull request by VeronikaNguyen**
- modified link for the algorithm instruction file
- added instructions for windows installation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ via the standard commands, i.e.,
 
 In addition, algorithms not denoted with "\*" above are not enabled for
 TLS operations. This designation can be changed by modifying the
-"enabled" flags in the main [algorithm configuration file](oqs-template/generate.yml)
+"enabled" flags in the main [algorithm configuration file](oqs-template/generate-extras.yml)
```
Comment (baentsch) [inline README.md:57]: This change is not correct; my bad to not have done #109 earlier: When that lands, the original reference is correct, so please revert this change.

File: README.md [hunk 2]
```diff
@@ -178,6 +178,71 @@ excludes all algorithms of the "Sphincs" family.
 *Note*: By default, interoperability testing with oqs-openssl111 is no longer
 performed by default but can be manually enabled in the script `scripts/runtests.sh`.
 
+Building under Windows
+--------------------
+The instructions below were tested on Windows 10 and 11 using MSYS2 MINGW64 (building with Visual Studio 2019 was NOT successful in our case). They are a bit experimental and only provisional. Note that the tests were skipped in the building process, however, setting up a test server and client with PQC algorithms still works.
```
Comment (baentsch) [inline README.md:183]: A statement like "in our case" reads weird if part of the project README. Could you please rephrase to state "...was not successful and still is an [open work item](https://github.com/open-quantum-safe/oqs-provider/issues/47]"? Also, any lessons learned (what went wrong, what should be looked into to get MSVC to work) would be a great addition to the comments section in https://github.com/open-quantum-safe/oqs-provider/issues/47.
Also regarding testing: If this doesn't work, can you state why? Is it something that can be fixed easily (.bat file instead of .sh, for example) or that is so complex that the platform should be excluded from testing completely, e.g., in https://github.com/open-quantum-safe/oqs-provider/blob/main/test/CMakeLists.txt?

File: README.md [hunk 3]
```diff
@@ -178,6 +178,71 @@ excludes all algorithms of the "Sphincs" family.
...
+
+    git clone https://github.com/open-quantum-safe/liboqs.git
+    cd liboqs
+    cmake -DCMAKE_INSTALL_PREFIX=$(pwd)/../.local -S . -B _build
+    cmake --build _build && cmake --install _build
+    cd ..
+
+Further `liboqs` build options are [documented here](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs).
+
+## Building the provider
```
Comment (baentsch) [inline README.md:220]: The following are actual code changes that would be good to have in the baseline code (if they don't break it): Does anything speak against adding them as-is to the files as documented?

File: README.md [hunk 4]
```diff
@@ -178,6 +178,71 @@ excludes all algorithms of the "Sphincs" family.
...
+Add `oqsprov.def` to the following section in CMakeLists.txt of the directory `oqsprov` in line 8.
+
+    set(PROVIDER_SOURCE_FILES
+        oqsprov.c oqsprov_capabilities.c oqsprov_keys.c
+        oqs_kmgmt.c oqs_sig.c oqs_kem.c
+        oqs_encode_key2any.c oqs_endecoder_common.c oqs_decode_der2key.c oqsprov_bio.c
+        oqsprov.def
+    )
+
+Modify `tests/test_common.c` in line 27 by replacing `while((comma = strchr(alglist, ','))) {` with :
```
Comment (baentsch) [inline README.md:236]: This is highly risky (line numbers change with code changes). Better would be a permalink, such as https://github.com/open-quantum-safe/oqs-provider/blob/5dc7df1506256f43a31b01d97110c10648f1a4d2/test/test_common.c#L27

That however makes me wonder: Which code base are you referring-to in this README change? It seems the code already does use `index` instead of `strchr`, thus making this modification suggestion moot, no? Or is the suggestion to use `strchr` instead of `index`?

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks again, @VeronikaNguyen for the PR: Some questions/comments inline. In general, could you please change it so that as few code changes as possible are documented to be needed? Those changes that are needed should be(come) part of the overall project (not breaking the current build, of course): That way, oqs-provider could make progress (towards full Windows support) and readers don't have to read  (or do) too much: In an ideal PR, the README does not have to be changed, but code changes are such that everything simply works on Windows (and Unix as before).

File: README.md (hunk 3 above)
Comment (VeronikaNguyen) [reply README.md:220]: Thanks for the constructive feedback. Since I am not too familiar with cmake and the building process, I was reluctant to add the following to the source code. To my knowledge it should work under Linux as well so I will change the source code.

**Comment by baentsch** [comment]
Thanks again for the contribution and improvement/putting the project "en route" to Windows support. Please feel free to add Windows CI support if you'd be comfortable with that/have time. Otherwise I'll give it a try (probably using AppVeyor) based on your instructions just added...
